hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThread.java

```diff
@@ -109,7 +109,7 @@ public final void addTask(Runnable task) {
      * @param task the task to add.
      * @throws NullPointerException if task is null
      */
-    public final void addTaskAndWakeup(Runnable task) {
+    public void addTaskAndWakeup(Runnable task) {
```

Comment:
Difficult. I think it is fine without adding a test here. Since it the same behavior as before + adds notification on top of that.